src/Hosting.Services/HostBuilderExtensions.cs

```diff
@@ -75,53 +77,39 @@ private static IHost BuildServiceFabricService<TRunner, TService, TContext>(
 		{
 			string serviceNameForLogging = serviceName;
 
-			try
+			if (string.IsNullOrWhiteSpace(serviceName))
 			{
-				if (string.IsNullOrWhiteSpace(serviceName))
-				{
-					// use executing assembly name for logging since application name not available
-					serviceNameForLogging = Assembly.GetExecutingAssembly().GetName().FullName;
-					throw new ArgumentException("Service type name is null of whitespace", nameof(serviceName));
-				}
-
-				builderAction(new ServiceFabricHostBuilder<TService, TContext>(builder));
-
-				IHost host = builder
-					.ConfigureServices((context, collection) =>
-					{
-						collection
-							.Configure<ServiceRegistratorOptions>(options =>
-							{
-								options.ServiceTypeName = serviceName;
-							})
-							.AddOmexServiceFabricDependencies<TContext>()
-							.AddSingleton<IOmexServiceRegistrator, TRunner>()
-							.AddHostedService<OmexHostedService>();
-					})
-					.UseDefaultServiceProvider(options =>
-					{
-						options.ValidateOnBuild = true;
-						options.ValidateScopes = true;
-					})
-#pragma warning disable OMEX188 // AddOmexLogging using OmexLogger is obsolete. DiagnosticId = "OMEX188"
-					.ConfigureLogging(builder => builder.AddOmexLogging())
-#pragma warning restore OMEX188 // AddOmexLogging using OmexLogger is obsolete. DiagnosticId = "OMEX188"
-					.Build();
+				// use executing assembly name for logging since application name not available
+				serviceNameForLogging = Assembly.GetExecutingAssembly().GetName().FullName;
+				throw new ArgumentException("Service type name is null of whitespace", nameof(serviceName));
+			}
 
-#pragma warning disable OMEX188 // InitializationLogger using OmexLogger is obsolete. DiagnosticId = "OMEX188"
-				InitializationLogger.LogInitializationSucceed(serviceNameForLogging);
-#pragma warning restore OMEX188 // InitializationLogger using OmexLogger is obsolete. DiagnosticId = "OMEX188"
+			builderAction(new ServiceFabricHostBuilder<TService, TContext>(builder));
 
-				return host;
-			}
-			catch (Exception e)
-			{
-#pragma warning disable OMEX188 // InitializationLogger using OmexLogger is obsolete. DiagnosticId = "OMEX188"
-				InitializationLogger.LogInitializationFail(serviceNameForLogging, e);
-#pragma warning restore OMEX188 // InitializationLogger using OmexLogger is obsolete. DiagnosticId = "OMEX188"
+			IHost host = builder
+				.ConfigureServices((context, collection) =>
+				{
+					collection
+						.Configure<ServiceRegistratorOptions>(options =>
+						{
+							options.ServiceTypeName = serviceName;
+						})
+						.AddOmexServiceFabricDependencies<TContext>()
+						.AddSingleton<IOmexServiceRegistrator, TRunner>()
+						.AddHostedService<OmexHostedService>();
+					// If specific context constructs are not registered, add defaults for telemetry purposes
+					collection.TryAddTransient<IServiceContext, EmptyServiceContext>();
+					collection.TryAddTransient<IExecutionContext, BaseExecutionContext>();
```

Comment:
Both interfaces already registered in .AddOmexServiceFabricDependencies()

Reply:
I suppose you later noticed that .AddOmexServiceFabricDependencies is called a couple of lines higher.